core-bundle/src/Resources/views/Frontend/preview_toolbar_base_js.html.twig

```diff
@@ -284,5 +284,26 @@
                 });
             });
         });
+
+        // copy published link to clipboard
+        window.addEventListener('cto-toolbar-loaded', function() {
+            const copyButton = document.getElementById('copyPublishedPath');
+            const input = document.getElementById('publishedPathInput');
+            input.value = window.location.href.replace('{{ preview_script }}', '');
+            copyButton.addEventListener('click', function(event) {
+                function clipboardFallback() {
+                    input.select();
+                    input.setSelectionRange(0, 99999);
+                    document.execCommand('copy');
+                }
+                event.preventDefault();
+                if (navigator.clipboard && navigator.clipboard.writeText) {
+                    navigator.clipboard.writeText(input.value).catch(clipboardFallback);
+                }
+                else {
+                    clipboardFallback();
```

Comment:
Why is this encapsulated in a function instead of just executed right here?

Reply:
Because it is used twice, see end of line 301.

Reply:
My bad. 🙈